src/impact-metrics/metric-client.ts

```diff
@@ -1,137 +1,12 @@
-  }
-
-  defineGauge(name: string, help: string) {
-    if (!name || !help) {
-      this.emit(UnleashEvents.Warn, `Gauge name or help cannot be empty: ${name}, ${help}.`);
-      return;
-    }
-    const labelNames = ['featureName', 'appName', 'environment'];
-    this.metricRegistry.gauge({ name, help, labelNames });
-  }
-
-  defineHistogram(name: string, help: string, buckets?: number[]) {
-    if (!name || !help) {
-      this.emit(UnleashEvents.Warn, `Histogram name or help cannot be empty: ${name}, ${help}.`);
-      return;
-    }
-    const labelNames = ['featureName', 'appName', 'environment'];
-    this.metricRegistry.histogram({ name, help, labelNames, buckets: buckets || [] });
-  }
-
-  private getFlagLabels(flagContext?: MetricFlagContext): MetricLabels {
-    let flagLabels: MetricLabels = {};
-    if (flagContext) {
-      for (const flag of flagContext.flagNames) {
-        const variant = this.variantResolver.forceGetVariant(flag, flagContext.context);
-
-        if (variant.enabled) {
-          flagLabels[flag] = variant.name;
-        } else if (variant.feature_enabled) {
-          flagLabels[flag] = 'enabled';
-        } else {
-          flagLabels[flag] = 'disabled';
-        }
-      }
-    }
-    return flagLabels;
-  }
-
-  incrementCounter(name: string, value?: number, flagContext?: MetricFlagContext): void {
-    const counter = this.metricRegistry.getCounter(name);
-    if (!counter) {
-      this.emit(
-        UnleashEvents.Warn,
-        `Counter ${name} not defined, this counter will not be incremented.`,
-      );
-      return;
-    }
-
-    const flagLabels = this.getFlagLabels(flagContext);
-
-    const labels = {
-      ...flagLabels,
-      ...this.staticContext,
-    };
-
-    counter.inc(value, labels);
-  }
-
-  updateGauge(name: string, value: number, flagContext?: MetricFlagContext): void {
-    const gauge = this.metricRegistry.getGauge(name);
-    if (!gauge) {
-      this.emit(UnleashEvents.Warn, `Gauge ${name} not defined, this gauge will not be updated.`);
-      return;
-    }
-
-    const flagLabels = this.getFlagLabels(flagContext);
-
-    const labels = {
-      ...flagLabels,
-      ...this.staticContext,
-    };
-
-    gauge.set(value, labels);
-  }
-
-  observeHistogram(name: string, value: number, flagContext?: MetricFlagContext): void {
-    const histogram = this.metricRegistry.getHistogram(name);
-    if (!histogram) {
-      this.emit(
-        UnleashEvents.Warn,
-        `Histogram ${name} not defined, this histogram will not be updated.`,
-      );
-      return;
-    }
-
-    const flagLabels = this.getFlagLabels(flagContext);
-
-    const labels = {
-      ...flagLabels,
-      ...this.staticContext,
-    };
-
-    histogram.observe(value, labels);
-  }
-}
+import { Unleash } from '../unleash';
 
 export class UnleashMetricClient extends Unleash {
-  public impactMetrics: MetricsAPI;
-
   constructor(...args: ConstructorParameters<typeof Unleash>) {
```

Comment:
Just so we can be polite to potential experimental users in the wild

Reply:
those users are us :) I appreciate the non breaking migration path so that we can update 2 repos (core + enterprise) at different intervals